opensquirrel/passes/validator/routing_validator.py

```diff
@@ -21,8 +21,9 @@ def validate(self, ir: IR) -> None:
         """
         non_executable_interactions = []
         for statement in ir.statements:
-            instruction: Instruction = cast("Instruction", statement)
-            args = instruction.arguments
+            if not isinstance(statement, Instruction):
+                continue
```

Comment:
```suggestion
            if not isinstance(statement, AsmDeclaration):
                continue
```
And obviously `from opensquirrel.ir import AsmDeclaration`

Reply:
Alright! I also just notices the suggestion I made was wrong to begin with :) so sorry about that